x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/store/RoleDescriptorStore.java

```diff
@@ -107,6 +108,19 @@ public void resolveApiKeyRoleReference(
             || (apiKeyRoleReference.getRoleType() == RoleReference.ApiKeyRoleType.LIMITED_BY
                 && rolesRetrievalResult.getRoleDescriptors().stream().noneMatch(RoleDescriptor::hasRestriction))
             : "there should be zero limited-by role descriptors with restriction and no more than one assigned";
+        // TODO we need unit tests for edge-cases here, for instance, we need to test the REST API keys are never checked for invalid legacy
```

Comment:
not sure we need a test that a method is not called. I added a bit of javadoc and the method and class are pretty self evident.